src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -1017,6 +1018,10 @@ public void setOpen(boolean isOpen) {
 
 	public boolean isOpen() {
 
+		if(TownySettings.isTownBankruptsyEnabled() && isBankrupt()) {
```

Comment:
Either change to `return (isBankrupt ? false : isOpen);`
Or preferably, move this check to within the TownCommand's parsing of the join command. Having it here would potentially mess with other plugins trying to do things differently using Towny's API.

Reply:
done
- I added the suggested logic